src/composite.c

```diff
@@ -0,0 +1,119 @@
+#include <stdio.h>
+#include <vips/vips.h>
+
+/*
+  Composite images `src` and `dst`
+ */
+int Composite(VipsObject *context, VipsImage *src, VipsImage *dst, VipsImage **out) {
+  const int ALPHA_BAND_INDEX = 3;
+  const int NUM_COLOR_BANDS = 3;
+
+  // These images will all be unreffed when context is unreffed:
+  VipsImage **t = (VipsImage **) vips_object_local_array(context, 3);
+
+  // Extract RGB bands:
+  VipsImage *srcRGB;
+  VipsImage *dstRGB;
+  if (vips_extract_band(src, &srcRGB, 0, "n", NUM_COLOR_BANDS, NULL) ||
+      vips_extract_band(dst, &dstRGB, 0, "n", NUM_COLOR_BANDS, NULL))
```

Comment:
This should probably use `sharp::HasAlpha` and it might be useful to add `sharp::GetAlpha` similar to `get_alpha` in https://github.com/jcupitt/ruby-vips/issues/28#issuecomment-9014826. 

Currently, we only have the need to composite RGBA PNGs for our purposes. Do you think we could land this PR like this or should we keep it open until any n-band images are correctly handled?